### DISCUSSION TITLE
Help with Custom Links

### DISCUSSION THREAD

**Discussion by cyberndj** [netbox-community/netbox#10548] [Q&A] [answered]
Howdy,
We have a custom Link that references a custom field on the device page to open a new page to our monitoring software (Solarwinds Orion). I would like to do some work with the button, based on items. We 

A) Hide (disable/not show) the button if no URL is configured
B) Change the color of the button based on the Orion status (traffic signal style

 For A), If the custom field is blank, the button shows up as black text (the cf.orion description text)/white background. If the field isn't black, it shows up as yellow button (as desired). 

For B) If jinja2 can do an API &| JSON lookup on field, how can we change the button color? (ie  If UP: button class=Green, ELSEIF Down: button class=Red, ELSE all others: button class=Yellow)

Currently:

Link Text:
{% if object.cf.orion|length %}
Open in Solarwinds
{% endif %}

Link URL
{% if object.cf.orion|length %}
{{ object.cf.orion }}
{% endif %}

**Comment by drmsoffall** [comment, marked as answer]
For A, instead of using 'length' try testing for 'is not none'.

**Reply by cyberndj** [reply]
Howdy,

That worked for A. 
for Ref, the updated line is:  {% if object.cf.orion is not none %}

**Reply by cyberndj** [reply]
For B,
It might be easier to split it into B1 and B2.
B1 is if/how to change the button color on a specific device and
B2 is to do the API/JSON lookup to get a value.

-Jake

**Comment by PVM-NL** [comment]
Hi cyberndj,

I was wondering how you did it with solarwinds?
As i am trying to use custom links to create links for site locations. But in Solarwinds it is listed in url as https://solarwinds.local/Orion/SummaryView.aspx?ViewName=Groups%20SummaryC:46.

Did you create a hidden custom field with url information in there and then a customlink on the object that is listed to solarwinds object?

**Comment by cyberndj** [comment]
Hey @PVM-NL,
TLDR; Yes as you described.

We have a custom field with the url to Orion for the device. We then have a custom link that uses the custom field that opens the page in the custom field. Hopefully the pics will help.

![image](https://user-images.githubusercontent.com/16530317/234331916-c6689831-b2c9-4bbf-a70b-b90c14e2747d.png)
![image](https://user-images.githubusercontent.com/16530317/234332073-91479b6c-a3ec-4733-8527-467d8bebdfd1.png)
